### PR TITLE
fixed exception when there is no conda lockfile

### PR DESCRIPTION
**Pull request by munishchouhan**
This pr will fix the exception, when there is no conda file in logs

### CODE REVIEW

File: src/main/groovy/io/seqera/wave/service/logs/BuildLogServiceImpl.groovy [hunk 1]
```diff
@@ -99,20 +99,21 @@ class BuildLogServiceImpl implements BuildLogService {
...
     void onBuildEvent(BuildEvent event) {
         if(event.result.logs) {
-            CompletableFuture.supplyAsync(() -> storeLog(event.result.buildId, event.result.logs), ioExecutor)
+            CompletableFuture.supplyAsync(() -> storeLog(event.result.buildId, event.result.logs, event.request.condaFile), ioExecutor)
         }
     }
 
     @Override
-    void storeLog(String buildId, String content){
+    void storeLog(String buildId, String content, String condaFile) {
```
Comment (pditommaso) [inline src/main/groovy/io/seqera/wave/service/logs/BuildLogServiceImpl.groovy:107]:
```suggestion
    void storeLog(String buildId, String content, boolean storeCondaLock) {
```

better to keep as bool

File: src/main/groovy/io/seqera/wave/service/logs/BuildLogServiceImpl.groovy [hunk 2]
```diff
@@ -99,20 +99,20 @@ class BuildLogServiceImpl implements BuildLogService {
     @EventListener
     void onBuildEvent(BuildEvent event) {
         if(event.result.logs) {
-            CompletableFuture.supplyAsync(() -> storeLog(event.result.buildId, event.result.logs, event.request.condaFile), ioExecutor)
+            CompletableFuture.supplyAsync(() -> storeLog(event.result.buildId, event.result.logs, event.request.condaFile != null), ioExecutor)
```
[the remark this answers is not in the document]
Comment (munishchouhan) [reply src/main/groovy/io/seqera/wave/service/logs/BuildLogServiceImpl.groovy:102]: but method is expecting boolean

`102: [Static type checking] - Cannot find matching method io.seqera.wave.service.logs.BuildLogServiceImpl#storeLog(java.lang.String, java.lang.String, java.lang.String).`

File: src/main/groovy/io/seqera/wave/service/logs/BuildLogServiceImpl.groovy (hunk 1 above)
Comment (munishchouhan) [reply src/main/groovy/io/seqera/wave/service/logs/BuildLogServiceImpl.groovy]: done

File: src/main/groovy/io/seqera/wave/service/logs/BuildLogServiceImpl.groovy (hunk 2 above)
[the remark this answers is not in the document]
Comment (pditommaso) [reply src/main/groovy/io/seqera/wave/service/logs/BuildLogServiceImpl.groovy:102]: let's ignore it then

[the remark this answers is not in the document]
Comment (munishchouhan) [reply src/main/groovy/io/seqera/wave/service/logs/BuildLogServiceImpl.groovy:102]: you mean, i should not check or use string in method?

[the remark this answers is not in the document]
Comment (munishchouhan) [reply src/main/groovy/io/seqera/wave/service/logs/BuildLogServiceImpl.groovy:102]: ok understood
ignore this change

**Comment by pditommaso** [comment]
What about this eb2f67af

**Comment by munishchouhan** [comment]
> What about this [eb2f67a](https://github.com/seqeralabs/wave/commit/eb2f67af50ba38890627a9c953a9c2b2ad680582)

this is already being checked in extractCondaLockFile, i was trying to avoid the log processing

https://github.com/seqeralabs/wave/blob/469c3082444cd008b61cd9d40526d73b247d2e89/src/main/groovy/io/seqera/wave/service/logs/BuildLogServiceImpl.groovy#L189-L194

**Comment by pditommaso** [comment]
That's check is cheap, better to keep the interface simpler and infer from the content

**Pull request merged**